includes/fs-essential-functions.php

```diff
@@ -36,6 +36,39 @@ function fs_normalize_path( $path ) {
 		}
 	}
 
+	if ( ! function_exists( 'fs_get_plugins' ) ) {
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.2.1
+         *
+         * @param string $plugin_folder
+         * @param bool   $delete_cache
+         *
+         * @return array
+         */
+	    function fs_get_plugins( $plugin_folder = '', $delete_cache = false ) {
+            if ( ! $cached_plugins = wp_cache_get( 'plugins', 'plugins' ) ) {
+                $cached_plugins = array();
+            }
+
+            if ( isset( $cached_plugins[ $plugin_folder ] ) ) {
+                $plugins = $cached_plugins[ $plugin_folder ];
+            } else {
+                if ( ! function_exists( 'get_plugins' ) ) {
+                    require_once ABSPATH . 'wp-admin/includes/plugin.php';
+                }
+
+                $plugins = get_plugins();

```

Comment:
@fajardoleo from what I've seen, `get_plugins()` is a resource intensive method, therefore, I suggest that before we delete the cache, first check if WooCommerce is installed at all. And only clear it if it is installed.